src/transport/encoding/raptorq.rs

```diff
@@ -19,42 +20,68 @@ pub(crate) use encoder::RaptorQEncoder;
 
 struct ChunkedPayload<'a>(&'a BroadcastPayload);
 
+// ObjectTransmissionInformation Size (Raptorq header)
+const TRANSMISSION_INFO_SIZE: usize = 12;
+
+// UID Size (Blake2s256)
+const UID_SIZE: usize = 32;
+
+// EncodingPacket min size (RaptorQ packet)
+const MIN_ENCODING_PACKET_SIZE: usize = 5;
+
+const MIN_CHUNKED_SIZE: usize =
+    UID_SIZE + TRANSMISSION_INFO_SIZE + MIN_ENCODING_PACKET_SIZE;
+
+impl<'a> TryFrom<&'a BroadcastPayload> for ChunkedPayload<'a> {
+    type Error = io::Error;
+    fn try_from(value: &'a BroadcastPayload) -> Result<Self, Self::Error> {
+        if value.gossip_frame.len() < MIN_CHUNKED_SIZE {
+            Err(io::Error::new(
+                ErrorKind::InvalidInput,
```

Comment:
If you want to keep the `io::Error` here, you should use `ErrorKind::UnexpectedEof` instead.